src/main/java/telraam/database/daos/LapDAO.java

```diff
@@ -29,6 +30,14 @@ public interface LapDAO extends DAO<Lap> {
     @RegisterBeanMapper(Lap.class)
     List<Lap> getAllBySourceSorted(@Bind("lapSourceId") Integer lapSourceId);
 
+    @SqlQuery("SELECT t.id as team_id, (SELECT COUNT(*) FROM lap WHERE lap_source_id = :lapSourceId AND timestamp <= :timestamp and team_id = t.id) as count FROM team t")
+    @RegisterBeanMapper(LapCount.class)
+    List<LapCount> getAllBeforeTime(@Bind("lapSourceId") Integer lapSourceId, @Bind("timestamp") Timestamp timestamp);
+
+    @SqlQuery("SELECT * FROM lap WHERE lap_source_id = :lapSourceId AND timestamp <= :timestamp AND team_id = :teamId")
+    @RegisterBeanMapper(Lap.class)
+    List<Lap> getAllForTeamBeforeTime(@Bind("lapSourceId") Integer lapSourceId, @Bind("teamId") Integer teamId, @Bind("timestamp") Timestamp timestamp);
```

Comment:
I would do this the same way as above, return the lap count here?
The result will have a redundant teamID, but the way of fetching will be the same, which is preferable I think?